---
title: "Applied Bioinformatics Day 3"
author: "Luca Gaessler"
date: "2025-10-08"
image: "Nextflow_scheme.png"
categories: [course, medbioinfo, nextflow]
description: "Hosted by Sweden's School for Medical Bioinformatics"
---

## Key Concepts I Learned Today

- **Nextflow**
- **nf-core**

### Nextflow
Nextflow is a tool used to connect different parts of your analysis. While this might trigger reactions like "I can run my scripts myself!" and "This is more a nice-to-have!", the use of workflow managers is real and multifold. Nextflow scripts not only lower manual efforts, and thereby reduce human error, but also enhance analytical scalability and reproducibility. When scripts are run over many samples, high-performance clusters are essential. When transferring scripts from local computers to clusters, Nexflow is a game-changer. It always keeps the functional logic of a workflow separate from its execution, meaning that you only have to exchange your local for appropriate server parameters to scale up your analysis. Additionally, Nextflow inherently incorporates version control and containerization, which - as I told about in the previous posts - are true boosters of science reproducibility. 

A very useful function in Nextflow is the "resume" option. Imagine your pipeline runs into an error right after a computationally expensive calculation. After fixing the bug, "resume" allows you to re-run the fixed pipeline from where it stopped, skipping all unchanged steps and recovering their results from cache instead. 

![**Basic Nextflow Architecture.** Nextflow processes, which contain the executable scripts, are connected via channels that move the data along the workflow. Processes are modular and can therefore be written in different programming languages. Adapted from https://carpentries-incubator.github.io/workflows-nextflow/aio.html (last visit: 08/10/2025)](Nextflow_scheme.png){#fig-id width=80% height=80% fig-align="center"}

### nf-core
Building up on Nextflow, nf-core is a web-based platform that provides curated & ready-to-use bioinformatic pipelines. The platform is largely community-driven. Volunteers create and validate the provided pipelines (at the time of this post, 139x pipelines are available) - the code is freely accessible on GitHub. Judging from my experience, most of the available pipelines are built for data processing tasks, while the focus lies less on downstream biological analyses. A very prominent use-case of nf-core is their "rnaseq" pipeline, which generates count matrices form raw FASTQ files, including quality control reports vias FastQC.

## Reflections
Although I currently see myself more as data scientist with a focus on biological analyses, workflow managers and off-the-shelf pipelines will definitely play an important part in the future of my PhD projects. Biological datasets are getting larger and larger, which calls for increased scalability and the need for high-performance computing clusters. All of this is facilitated by tools like Nextflow and will definitely change the way I work right now.

## Next Steps
While we have only scratched the surface of bioinformatic pipelines today, tomorrow will bring more hands-on tasks on Nextflow as well as the use of nf-core pipelines.

**Thanks for reading!**